File: array_Reverse.rb
#!/bin/ruby

n = gets.strip.to_i
arr = gets.strip
arr = arr.split(' ').map(&:to_i)

b = arr.reverse
arr.each{|x| print x, " "}
puts "\n"
b.each{|x| print x, " "}

puts "\nUsing negative index from -1..1"
c = arr[-1..1]
b.each{|x| print x, " "}

puts "\nUsing ! will reverse in-line"
arr.reverse!
arr.each{|x| print x, " "}

File: array_Reverse.py
#!/bin/python

import sys


n = int(raw_input().strip())
arr = map(int,raw_input().strip().split(' '))

print '\n'
print "Using Array index from -1"
b = arr[::-1]
for i in b:
    print i,
print '\n'

print "Using reversed method"
for i in reversed(arr):
    print str(i) ,
print '\n'

print "Using Array reverse method"
arr.reverse()
for i in arr:
    print i,